File: SIMPLEAPI_Demo/SIMPLEAPI_Demo/Impresion/Core/Classes/FontInfo.cs
using System;
using System.Drawing;

namespace SIMPLEAPI_Demo.Impresion.Core.Classes
{
    public static class FontInfo
    {
        public const string FontName = "Consolas"; // if changed, change max chars per font size table in method GetMaxChars(font)

        public static int GetMaxChars(int fontSize, FontStyle fontStyle, bool testing = false)
        {
            return GetMaxChars(new Font(FontName, fontSize, fontStyle), testing);
        }
        public static int GetMaxChars(Font font, bool testing = false)
        {
            if (testing)
                return 100;

            if (font.FontFamily.Name != FontName)
                throw new Exception("Font not supported");

            int fontSize = (int)font.Size;

            if (font.Style == FontStyle.Bold)
            {
                if (fontSize == 7) return 52;
                else if (fontSize == 8) return 45;
                else if (fontSize == 10) return 36;
                else if (fontSize == 12) return 15;
                else
                    throw new Exception("fontSize not permitted");
            }
            else if (font.Style == FontStyle.Italic)
            {
                if (fontSize == 7) return 52;
                else if (fontSize == 8) return 45;
                else if (fontSize == 10) return 36;
                else if (fontSize == 12) return 15;
                else
                    throw new Exception("fontSize not permitted");
            }
            else
            {
                if (fontSize == 7) return 52;
                else if (fontSize == 8) return 45;
                else if (fontSize == 10) return 36;
                else if (fontSize == 12) return 15;
                else
                    throw new Exception("fontSize not permitted");
            }
        }
    }
}

File: SIMPLEAPI_Demo/SIMPLEAPI_Demo_VB/Impresion/Core/Classes/FontInfo.vb
Public Class FontInfo
    Public Const FontName As String = "Consolas"
    '// if changed, change max chars per font size table in method GetMaxChars(font)
    Public Shared Function GetMaxChars(ByVal fontSize As Integer, ByVal fontStyle As FontStyle, ByVal Optional testing As Boolean = False) As Integer
        Return GetMaxChars(New Font(FontName, fontSize, fontStyle), testing)
    End Function

    Public Shared Function GetMaxChars(ByVal font As Font, ByVal Optional testing As Boolean = False) As Integer
        If testing Then Return 100
        If font.FontFamily.Name <> FontName Then Throw New Exception("Font not supported")
        Dim fontSize As Integer = CInt(font.Size)

        If font.Style = FontStyle.Bold Then

            If fontSize = 7 Then
                Return 52
            ElseIf fontSize = 8 Then
                Return 45
            ElseIf fontSize = 10 Then
                Return 36
            ElseIf fontSize = 12 Then
                Return 15
            Else
                Throw New Exception("fontSize not permitted")
            End If
        ElseIf font.Style = FontStyle.Italic Then

            If fontSize = 7 Then
                Return 52
            ElseIf fontSize = 8 Then
                Return 45
            ElseIf fontSize = 10 Then
                Return 36
            ElseIf fontSize = 12 Then
                Return 15
            Else
                Throw New Exception("fontSize not permitted")
            End If
        Else

            If fontSize = 7 Then
                Return 52
            ElseIf fontSize = 8 Then
                Return 45
            ElseIf fontSize = 10 Then
                Return 36
            ElseIf fontSize = 12 Then
                Return 15
            Else
                Throw New Exception("fontSize not permitted")
            End If
        End If
    End Function
End Class
